src/main/java/org/sonarlint/intellij/core/server/events/ServerEventsService.kt

```diff
@@ -85,17 +92,30 @@ class ServerEventsProductionService : ServerEventsService {
                 project, TaintVulnerabilitiesPresenter::class.java
             ).presentTaintVulnerabilitiesForOpenFiles()
         }
-        ProjectManager.getInstance().openProjects.forEach { project ->
-            getService(project, SecurityHotspotsPresenter::class.java)
-                .presentSecurityHotspotsForOpenFiles()
+
+        identifyProjectsImpactedByIssueOrHotspotEvent(event).forEach { project ->
+            val impactedFiles = ArrayList<VirtualFile>()
+            val filePath = (event as ServerHotspotEvent).filePath
+            ProjectRootManager.getInstance(project).contentRoots.forEach {
+                if (it.isDirectory) {
+                    val matchedFile = it.findFileByRelativePath(filePath)
+                    if (matchedFile != null) {
+                        impactedFiles.add(matchedFile)
+                    }
+                } else {
+                    if (it.path.endsWith(filePath)) {
+                        impactedFiles.add(it)
+                    }
+                }
+            }
+            getService(project, AnalysisSubmitter::class.java).autoAnalyzeFiles(impactedFiles, TriggerType.SERVER_SENT_EVENT)
         }
     }
 
     private fun identifyProjectsImpactedByTaintEvent(event: ServerEvent): Set<Project> {
         val projectKey = when (event) {
             is TaintVulnerabilityRaisedEvent -> event.projectKey
             is TaintVulnerabilityClosedEvent -> event.projectKey
-            is IssueChangedEvent -> event.projectKey
```

Comment:
@damien-urruty-sonarsource what was the reason to put `IssueChangedEvent` along with the taint vulnerability events?